Code/src/main/java/nl/utwente/group10/ui/BackendUtils.java

```diff
@@ -31,4 +33,18 @@ public static Type dive(ConstT function, int steps) {
         }
         return function;
     }
+    
+    /**
+     * @return Whether or not t1 and t2 unify with each other.
```

Comment:
That's not really what this does. Instead, it *changes* t1 to match t2 (or vice versa). It will also recursively change other (seemingly unrelated) things if that's what's needed to make that happen. If there's no way to make it happen, it might *still* change t1 and t2, but it will also throw a type error/return false.

Please, *please* talk with @jjkester or at least look at the diagram I sent you. He will probably also tell you you probably shouldn't call unify from the frontend directly.